packages/airnode-adapter/src/response-processing/encoding.ts

```diff
@@ -1,66 +1,8 @@
 import { ethers } from 'ethers';
-import { parseArrayType } from './array-type';
 import { ResponseType, ValueType } from '../types';
 
-function convertUnsignedInteger(value: string) {
-  return ethers.utils.defaultAbiCoder.encode(['uint256'], [value]);
-}
-
-function convertSignedInteger(value: string) {
-  return ethers.utils.defaultAbiCoder.encode(['int256'], [value]);
-}
-
-function convertBool(value: boolean) {
-  return ethers.utils.defaultAbiCoder.encode(['bool'], [value]);
-}
-
-function convertBytes32(value: string) {
-  // We can't encode strings longer than 31 characters to bytes32.
-  // Ethers need to keep room for null termination
-  const trimmedValue = value.length > 31 ? value.substring(0, 31) : value;
-
-  const bytes32String = ethers.utils.formatBytes32String(trimmedValue);
-  return ethers.utils.defaultAbiCoder.encode(['bytes32'], [bytes32String]);
-}
-
-function convertAddress(value: string) {
-  return ethers.utils.defaultAbiCoder.encode(['address'], [value]);
-}
-
-function convertBytes(value: string | ArrayLike<number>) {
-  return ethers.utils.defaultAbiCoder.encode(['bytes'], [value]);
-}
-
-function convertString(value: string) {
-  return ethers.utils.defaultAbiCoder.encode(['string'], [value]);
-}
-
-function convertArray(value: ValueType, type: ResponseType): string {
-  return ethers.utils.defaultAbiCoder.encode([type], [value]);
-}
-
 export function encodeValue(value: ValueType, type: ResponseType): string {
-  const parsedArrayType = parseArrayType(type);
-  if (parsedArrayType) {
-    return convertArray(value, type);
-  }
+  if (type === 'bytes32') return ethers.utils.formatBytes32String(value as string);
```

Comment:
I have mixed feelings about this... There are two stages - casting the value and encoding. The logic works like this: 
- if the type is `bytes32`, the casting will trim the value to at most 31 characters, encoding is done by `formatBytes32String`
- if the type is `string`, casting will just stringify and encoding is done by `defaultAbiCoder` (line below)
- if the type is `bytes` casting will verify that the string is a valid hex string (e.g. `0x7468697320697320616e206578616d706c6520737472696e672074686174206973206120626974206c6f6e676572`) and again encode using `defaultAbiCoder`

Result:
- The handling is very different for `bytes32` and `bytes` (former expects string, latter "encoded" hex string)
- If I implemented the `bytes` to expect string and encode it the same way - I'd be essentially doing the same as what `string` does

WDYT should be the correct approach?

Reply:
@andreogle @bbenligiray

Reply:
In the previous version, there was only one way to return a string from the API response (encoded in `bytes32`), which is why we did it this way. This is confusing for the user but can be worked around https://api3workspace.slack.com/archives/C02AYRX8D89/p1633444784102900

Here, we definitely want to treat `bytes` as raw bytes and not an encoded string (because there is no other way to return raw bytes from the API and we want that). I guess one question is do we still treat `bytes32` as an encoded string (which is not obvious) or should we also treat `bytes32` as `bytes32` and tell people to use `string` if they want to return a string.

Reply:
Responded in Slack, but this:

> should we also treat bytes32 as bytes32 (so break backwards compatibility for it to make more sense) and tell people to use string if they want to return a string (which will actually be more expensive than bytes32).

seems like the correct thing to do to me?

Reply:
I had also an idea bout introducing "made up" `string32` which would encode the same way as `bytes32` currently.

Reply:
I am resolving this thread -let's continue the discussion in slack